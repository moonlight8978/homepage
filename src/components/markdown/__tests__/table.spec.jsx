import React from 'react'
import { render } from '@testing-library/react'

import Table, { TableRow, TableCell } from '../table'

const renderTable = children =>
  render(
    <table>
      <tbody>{children}</tbody>
    </table>
  )

describe('Table', () => {
  test('renders correctly', () => {
    const children = (
      <tbody>
        <tr>
          <td>Row</td>
        </tr>
      </tbody>
    )
    const { container } = render(<Table>{children}</Table>)
    expect(container.firstChild).toMatchSnapshot()
  })
})

describe('TableRow', () => {
  test('renders correctly', () => {
    const children = <td>Row</td>
    const { container } = renderTable(<TableRow>{children}</TableRow>)
    expect(container.firstChild).toMatchSnapshot()
  })
})

describe('TableHeader', () => {
  test('renders left-aligned header correctly', () => {
    const children = <>Row</>
    const { container } = renderTable(
      <tr>
        <TableCell isHeader>{children}</TableCell>
      </tr>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('renders center-aligned header correctly', () => {
    const children = <>Row</>
    const { container } = renderTable(
      <tr>
        <TableCell isHeader align="center">
          {children}
        </TableCell>
      </tr>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('renders right-aligned header correctly', () => {
    const children = <>Row</>
    const { container } = renderTable(
      <tr>
        <TableCell isHeader align="right">
          {children}
        </TableCell>
      </tr>
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})

describe('TableCell', () => {
  test('renders left-aligned cell correctly', () => {
    const children = <>Row</>
    const { container } = renderTable(
      <tr>
        <TableCell>{children}</TableCell>
      </tr>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('renders center-aligned cell correctly', () => {
    const children = <>Row</>
    const { container } = renderTable(
      <tr>
        <TableCell align="center">{children}</TableCell>
      </tr>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('renders right-aligned cell correctly', () => {
    const children = <>Row</>
    const { container } = renderTable(
      <tr>
        <TableCell align="right">{children}</TableCell>
      </tr>
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
