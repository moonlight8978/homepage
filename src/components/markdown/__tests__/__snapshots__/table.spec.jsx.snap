// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders correctly 1`] = `
<table
  class="table"
>
  <tbody>
    <tr>
      <td>
        Row
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`TableCell renders center-aligned cell correctly 1`] = `
<table>
  <tbody>
    <tr>
      <td
        align="center"
        class="cell"
      >
        Row
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`TableCell renders left-aligned cell correctly 1`] = `
<table>
  <tbody>
    <tr>
      <td
        align="left"
        class="cell"
      >
        Row
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`TableCell renders right-aligned cell correctly 1`] = `
<table>
  <tbody>
    <tr>
      <td
        align="right"
        class="cell"
      >
        Row
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`TableHeader renders center-aligned header correctly 1`] = `
<table>
  <tbody>
    <tr>
      <th
        align="center"
        class="cell header"
      >
        Row
      </th>
    </tr>
  </tbody>
</table>
`;

exports[`TableHeader renders left-aligned header correctly 1`] = `
<table>
  <tbody>
    <tr>
      <th
        align="left"
        class="cell header"
      >
        Row
      </th>
    </tr>
  </tbody>
</table>
`;

exports[`TableHeader renders right-aligned header correctly 1`] = `
<table>
  <tbody>
    <tr>
      <th
        align="right"
        class="cell header"
      >
        Row
      </th>
    </tr>
  </tbody>
</table>
`;

exports[`TableRow renders correctly 1`] = `
<table>
  <tbody>
    <tr
      class="row"
    >
      <td>
        Row
      </td>
    </tr>
  </tbody>
</table>
`;
